Platform/AlderlakeBoardPkg/Library/FspmUpdUpdateLib/FspmUpdUpdateLib.c

```diff
@@ -266,33 +269,54 @@ UpdateFspConfig (
-  Fspmcfg->MemorySpdPtr011  = SpdData[MemCfgData->SpdDataSel011];
-  Fspmcfg->MemorySpdPtr020  = SpdData[MemCfgData->SpdDataSel020];
-  Fspmcfg->MemorySpdPtr021  = SpdData[MemCfgData->SpdDataSel021];
-  Fspmcfg->MemorySpdPtr030  = SpdData[MemCfgData->SpdDataSel030];
-  Fspmcfg->MemorySpdPtr031  = SpdData[MemCfgData->SpdDataSel031];
-  Fspmcfg->MemorySpdPtr100  = SpdData[MemCfgData->SpdDataSel100];
-  Fspmcfg->MemorySpdPtr101  = SpdData[MemCfgData->SpdDataSel101];
-  Fspmcfg->MemorySpdPtr110  = SpdData[MemCfgData->SpdDataSel110];
-  Fspmcfg->MemorySpdPtr111  = SpdData[MemCfgData->SpdDataSel111];
-  Fspmcfg->MemorySpdPtr120  = SpdData[MemCfgData->SpdDataSel120];
-  Fspmcfg->MemorySpdPtr121  = SpdData[MemCfgData->SpdDataSel121];
-  Fspmcfg->MemorySpdPtr130  = SpdData[MemCfgData->SpdDataSel130];
-  Fspmcfg->MemorySpdPtr131  = SpdData[MemCfgData->SpdDataSel131];
+  MemId = PcdGet8 (PcdMemConfigId);
+
+  // Memory SPD Data.
+  if ((MemId >= RplpLp5AutoMicron4GMT62F1G32D2DSSpd) && (MemId <= RplpLp5AutoSamsung16GbK3KLALA0DMSpd)) {
+    //
+    // This needs to be kept in sync with the actual hardware
+    // configuration. Assigning SPD Data to Controller/Channel/Dimm
+    // that does not exist results in MRC code detecting a CRC error and
+    // retriggering MRC training.
+    //
+    Fspmcfg->MemorySpdPtr000  = (UINT32)(UINTN)RplpAutoLp5SpdConfigs[MemId]->SpdData;
+    Fspmcfg->MemorySpdPtr010  = (UINT32)(UINTN)RplpAutoLp5SpdConfigs[MemId]->SpdData;
+    Fspmcfg->MemorySpdPtr020  = (UINT32)(UINTN)RplpAutoLp5SpdConfigs[MemId]->SpdData;
+    Fspmcfg->MemorySpdPtr030  = (UINT32)(UINTN)RplpAutoLp5SpdConfigs[MemId]->SpdData;
+    Fspmcfg->MemorySpdPtr100  = (UINT32)(UINTN)RplpAutoLp5SpdConfigs[MemId]->SpdData;
+    Fspmcfg->MemorySpdPtr110  = (UINT32)(UINTN)RplpAutoLp5SpdConfigs[MemId]->SpdData;
+    Fspmcfg->MemorySpdPtr120  = (UINT32)(UINTN)RplpAutoLp5SpdConfigs[MemId]->SpdData;
+    Fspmcfg->MemorySpdPtr130  = (UINT32)(UINTN)RplpAutoLp5SpdConfigs[MemId]->SpdData;
+  } else {
+    SpdData[0] = 0;
+    SpdData[1] = (UINT32)(UINTN) (((MEM_SPD0_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD0_TAG))->MemorySpdPtr0);
+    SpdData[2] = (UINT32)(UINTN) (((MEM_SPD1_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD1_TAG))->MemorySpdPtr1);
+    SpdData[3] = (UINT32)(UINTN) (((MEM_SPD2_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD2_TAG))->MemorySpdPtr2);
+    SpdData[4] = (UINT32)(UINTN) (((MEM_SPD3_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD3_TAG))->MemorySpdPtr3);
+    SpdData[5] = (UINT32)(UINTN) (((MEM_SPD4_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD4_TAG))->MemorySpdPtr4);
+    SpdData[6] = (UINT32)(UINTN) (((MEM_SPD5_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD5_TAG))->MemorySpdPtr5);
+    SpdData[7] = (UINT32)(UINTN) (((MEM_SPD6_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD6_TAG))->MemorySpdPtr6);
+    SpdData[8] = (UINT32)(UINTN) (((MEM_SPD7_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD7_TAG))->MemorySpdPtr7);
+#if defined(PLATFORM_RPLA)
```

Comment:
I think this #ifdef PLATFORM_RPLA ... #endif is wrong because the baseline code sets SpdData[9] unconditionally (cf. original line 279).

Reply:
Done.